randovania/games/dread/patcher/open_dread_patcher.py

```diff
@@ -131,29 +145,77 @@ def _pickup_detail_for_target(detail: ExportedPickupDetails) -> Optional[dict]:
                 model_name = "itemsphere"
             else:
                 model_name = detail.model.name
+            
+            ammoconfig = configuration.ammo_configuration.items_state
+            pbammo = item_db.ammo["Power Bomb Expansion"]
+
+            def get_resource(res: ConditionalResources) -> dict:
+                item_id = "ITEM_NONE"
+                quantity = 1
+                ids = [_get_item_id_for_item(r) for r, q in res.resources]
+                for r, q in res.resources:
+                    try:
+                        item_id = _get_item_id_for_item(r)
+                        quantity = q
+                        break
+                    except KeyError:
+                        continue
+                
+                if "ITEM_WEAPON_POWER_BOMB" in ids:
+                    item_id = "ITEM_WEAPON_POWER_BOMB"
+                
+                # non-required mains
+                if (item_id == "ITEM_WEAPON_POWER_BOMB_MAX"
+                    and not ammoconfig[pbammo].requires_major_item):
+                    item_id = "ITEM_WEAPON_POWER_BOMB"
+
+                return {"item_id": item_id, "quantity": quantity}
+            
+            # ugly hack
+            resources = [get_resource(res) for res in detail.conditional_resources]
+            if resources[0]["item_id"] == "ITEM_WEAPON_POWER_BOMB_MAX":
+                resources = [resources[-1]]
+
+            pickup_node = db.world_list.node_from_pickup_index(detail.index)
+            pickup_type = pickup_node.extra.get("pickup_type", "actor")
+
+            hud_text = detail.hud_text[0]
+            if len(detail.hud_text) > 1:
+                mapping = {
+                    "Varia Suit acquired.": "Progressive Suit acquired.",
```

Comment:
... there's _definitely_ a better way to do this 🤔

Reply:
I couldn't find how echoes does it so I just did the first thing I knew would work

Reply:
Echoes has a different text depending on which one you get.

My thought would be to add the original `PickupEntry` to `ExportedPickupDetails`, and then `if len(detail.hud_text) > 1` you do `f"{pickup.name} acquired."`.